jbmc/src/java_bytecode/java_entry_point.cpp

```diff
@@ -317,17 +319,89 @@ exprt::operandst java_build_arguments(
 
     parameters.function_id = goto_functionst::entry_point();
 
+    namespacet ns(symbol_table);
+
     // generate code to allocate and non-deterministicaly initialize the
-    // argument
-    main_arguments[param_number] = object_factory(
-      p.type(),
-      base_name,
-      init_code,
-      symbol_table,
-      parameters,
-      allocation_typet::LOCAL,
-      function.location,
-      pointer_type_selector);
+    // argument, if the argument has different possible object types (from casts
```

Comment:
The comment `from casts in the function body` sounds more like a possible heuristics than a necessity.